node/src/lib/deploy-utils.js

```diff
@@ -127,19 +127,33 @@ class DeployUtils {
     return doneSteps;
   }
 
-  async pollDeploymentStatus(deploymentLogId, shouldProcessDeploymentSteps = true) {
+  async pollDeploymentStatus(deployment, shouldProcessDeploymentSteps = true) {
     const MAX_TIME_IN_SECONDS = 10800; // 3 hours
     const start = Date.now();
     const stepsAlreadyLogged = [];
+    let previousStatus;
+
+    const pollableStatuses = ['IN_PROGRESS'];
+    if (deployment.status === 'QUEUED') {
+      pollableStatuses.push('QUEUED');
+      logger.info(`Deployment is queued! Waiting for it to start...`);
+      logger.info('Note: You can always stop waiting by using Ctrl+C');
+    }
 
     while (true) {
-      const { status } = await apiClient.callApi('get', `environments/deployments/${deploymentLogId}`);
+      const { type, status } = await this.getDeployment(deployment.id);
+
+      status === 'QUEUED' && logger.info('Waiting for deployment to start....');
```

Comment:
1. Please add a proper condition when not circuiting assignment  
2. Consider adding a message as to what are we waiting for again -

```suggestion

      if(status === 'QUEUED) logger.info('Queued deployment is still waiting for earlier deployments to finish...');
```